feat(@astrojs/cloudflare): add runtime support to `astro dev` (#8426)
withastro/astro: packages/integrations/cloudflare/README.md

## Access to the Cloudflare runtime

> You can access all the Cloudflare bindings and environment variables from Astro components and API routes through `Astro.locals`. Usually you need to use [Wrangler](https://github.com/cloudflare/workers-sdk) to be able to access the runtime.

Reviewer: ```suggestion
You can opt in to access all the Cloudflare bindings and environment variables from Astro components and API routes through `Astro.locals`. This replaces using [Wrangler](https://github.com/cloudflare/workers-sdk) to access the runtime.
```
I think this needs to be clearer, because the above phrase is a bit sophisticated English usage that requires familiarity with the two parts of that common structure, and only implies the (important) second half. It *could* potentially be interpreted as "Usually, you WILL (present tense! You will! this is our guidance!) use Wrangler to do this" NOT as you intended, "Usually you'd use Wrangler, **but now you don't have to!**") 

You could also add on to the existing sentence something that is explicitly clear, 

> ... to access the runtime, but configuring `cloudflare.runtime` will allow you to use Astro's dev server instead."

Author: While I concur with your comment, the technical implementation requires an additional PR to align with it. For now, I've removed the reference to wrangler in this section, as the other part of the Readme already contains all the necessary information.

}
```

> ### Astro Dev Server
> 
> The Adapter allows an opt-in flag, to enable the use of Environment Variables and the Cloudflare Request Object to be populated by the Astro Dev Server. There is no need to use wrangler in this case. This flag is `off` by default.
> 
> There are two supported modes: `local` & `remote`. In the mode `local` environmental variables are available, but the request object is populated from a static placeholder value. In the mode `remote` environmental variables and the a live fetched request object are available.

Reviewer: ```suggestion
### `cloudflare.runtime` 

`runtime: "off" | "local" | "remote"`
default `"off"`

This optional flag enables the Astro dev server to populate environment variables and the Cloudflare Request Object, avoiding the need for Wrangler.

-  `local`: environmental variables are available, but the request object is populated from a static placeholder value. 
- `remote`: environmental variables and the live, fetched request object are available.
- `off`: the Astro dev server will populate neither environment variables nor the request object. Use Wrangler to access Cloudflare bindings and environment variables.
```

I don't think "dev server" is the key idea here, it's the config object. I took a stab at this from the structure of the `mode` section above. Please make this true/in the right jargon, but I think this kind of structure will work well here!

}
```

### `cloudflare.runtime` 

`runtime: "off" | "local" | "remote"`
default `"off"`

This optional flag enables the Astro dev server to populate environment variables and the Cloudflare Request Object, avoiding the need for Wrangler.

> -  `local`: environmental variables are available, but the request object is populated from a static placeholder value. 
> - `remote`: environmental variables and the live, fetched request object are available.

Reviewer: ```suggestion
-  `local`: environment variables are available, but the request object is populated from a static placeholder value. 
- `remote`: environment variables and the live, fetched request object are available.
```
Ignore if "environmental" really is the right word, we just don't use it anywhere else so I'm double checking!

Author: Good catch ❤️
